app/models/build.rb

```diff
@@ -69,6 +69,31 @@ def file_from_repo(path, ttl: 1.hour)
     nil
   end
 
+  def import_kubernetes_roles
+    # TODO : Build -> KubernetesRoles association
+    project.roles.destroy_all
+
+    config_files = project.repository.directory_contents(git_sha, path: 'kubernetes')
+
+    config_files.map do |file|
+      file_contents = project.repository.file_contents(git_sha, file)
+      config_file = Kubernetes::RoleConfigFile.new(file_contents, file)
```

Comment:
You made a good point regarding additional files in that directory. I can filter the output to get only the .yaml or .json files. Thanks for the heads up.

Regarding the validation, I was thinking about that, but I'm still not sure about where and how to do it, i.e., if we should validate the files contents or validate the required data after parsing the file. Or... validate the files when creating the build.

We can address this is multiple ways, I'm just not sure about how to do it, yet.